tests/bears/BearTest.py

```diff
@@ -537,6 +560,126 @@ def test_is_debugged(self):
         section.append(Setting('debug_bears', 'abc, xyz'))
         self.assertEqual(_is_debugged(uut), False)
 
+    @patch('cProfile.Profile.dump_stats')
+    def test_profiler_with_no_directory_exists(self, dump_stats):
+        args = ()
+        kwargs = {}
+        section = Section('name')
+        section.append(Setting('profile', 'tRuE'))
+        bear = TestTwoBear(section, self.queue)
+        self.assertEqual(bear.run_bear_from_section(args, kwargs), [1, 2, 3])
+        dump_stats.assert_called_once_with(join(os.getcwd(),
+                                                'name_TestTwoBear.prof'))
+        section.append(Setting('profile', 'abc'))
+        bear = TestTwoBear(section, self.queue)
+        self.assertEqual(bear.run_bear_from_section(args, kwargs), [1, 2, 3])
+        dump_stats.assert_called_with(os.path.join(
+            bear.profile, 'name_TestTwoBear.prof'))
+        os.rmdir('abc')
+        section.append(Setting('profile', '1'))
+        bear = TestThreeBear(section, self.queue)
+        self.assertIsNone(bear.run_bear_from_section(args, kwargs))
+        dump_stats.assert_called_with(join(os.getcwd(),
+                                           'name_TestThreeBear.prof'))
+
+    @patch('cProfile.Profile.dump_stats')
+    def test_profiler_with_directory_exists(self, dump_stats):
+        args = ()
+        kwargs = {}
+        section = Section('name')
+        with TemporaryDirectory() as temp_dir:
+            section.append(Setting('profile', temp_dir))
+            bear = TestTwoBear(section, self.queue)
+            self.assertEqual(bear.run_bear_from_section(args, kwargs),
+                             [1, 2, 3])
+            dump_stats.assert_called_once_with(os.path.join(
+                bear.profile, 'name_TestTwoBear.prof'))
+
+    def test_profiler_with_file_path(self):
+        args = ()
+        kwargs = {}
+        section = Section('name')
+        with NamedTemporaryFile() as temp_file:
+            section.append(Setting('profile', temp_file.name))
+            bear = TestTwoBear(section, self.queue)
+            with self.assertRaisesRegex(SystemExit, '2'):
```

Comment:
assertRaisesRegex is inappropriate with SystemExit .  check the exit code, which is `e.code` iirc